Gemfile

```diff
@@ -1,72 +1,99 @@
-end
-
-if ENV['FACTER_GEM_VERSION']
-  gem "facter", *location_for(ENV['FACTER_GEM_VERSION'] || '~> 1.6')
-else
-  gem "facter", :require => false
-end
-
-if ENV['HIERA_GEM_VERSION']
-  gem "hiera", *location_for(ENV['HIERA_GEM_VERISON'])
-else
-  gem "hiera"
-end
-
-is_ruby18 = RUBY_VERSION.start_with? '1.8'
-
-group :development, :test do
-  if is_ruby18
-    gem 'pry', "~> 0.9.12"
+# Find a location or specific version for a gem. place_or_version can be a
+# version, which is most often used. It can also be git, which is specified as
+# `git://somewhere.git#branch`. You can also use a file source location, which
+# is specified as `file://some/location/on/disk`.
+def location_for(place_or_version, fake_version = nil)
+  if place_or_version =~ /^(git[:@][^#]*)#(.*)/
+    [fake_version, { :git => $1, :branch => $2, :require => false }].compact
+  elsif place_or_version =~ /^file:\/\/(.*)/
+    ['>= 0', { :path => File.expand_path($1), :require => false }]
   else
-    gem 'pry'
+    [place_or_version, { :require => false }]
   end
-  gem 'rake'
-  gem 'rspec', "~> 2.13.0"
-  gem 'mocha', "~> 0.10.5"
-  gem 'puppetlabs_spec_helper'
-  gem 'nokogiri', "~> 1.5.10"
-  gem 'mime-types', '<2.0',      :require => false
-  gem 'puppet-lint',             :require => false
-  gem 'simplecov',               :require => false
 end
 
-beaker_version = ENV['BEAKER_VERSION']
+group :development do
+  gem 'rake',                    :require => false
+  gem 'rspec', '~>2.14.0',       :require => false
+  gem 'puppet-lint',             :require => false
+  gem 'puppetlabs_spec_helper',  :require => false
+  gem 'puppet_facts',            :require => false
+  gem 'mocha', '~>0.10.5',       :require => false
+  gem 'nokogiri', '~>1.5.10',    :require => false, :platforms => :ruby
+  gem 'mime-types', '<2.0',      :require => false
+end
 group :system_tests do
+  beaker_version = ENV['BEAKER_VERSION'] || '~> 2.2'
   if beaker_version
     gem 'beaker', *location_for(beaker_version)
   else
-    gem 'beaker',                :require => false, :platforms => :ruby
+    gem 'beaker', :require => false, :platforms => :ruby
   end
-  gem 'beaker-puppet_install_helper', :require => false
+  gem 'beaker-puppet_install_helper',  :require => false
 end
 
-# see http://projects.puppetlabs.com/issues/21698
-platforms :mswin, :mingw do
-  gem "ffi", "~> 1.9.3", :require => false
-  gem "sys-admin", "~> 1.5.6", :require => false
-  gem "win32-dir", "~> 0.3.7", :require => false
-  gem "win32-process", "~> 0.6.5", :require => false
-  gem "win32-service", "~> 0.7.2", :require => false
-  gem "win32-taskscheduler", "~> 0.2.2", :require => false
+# The recommendation is for PROJECT_GEM_VERSION, although there are older ways
+# of referencing these. Add them all for compatibility reasons. We'll remove
+# later when no issues are known. We'll prefer them in the right order.
+puppetversion = ENV['PUPPET_GEM_VERSION'] || ENV['GEM_PUPPET_VERSION'] || ENV['PUPPET_LOCATION'] || '>= 0'
```

Comment:
It does that as a default. If the tests are passing on v4, why is this important? We should be moving everything forward to v4 nowadays. 

Or are you mentioning this as not happy *temporarily*?